### PR TITLE
stan rb: subscriber impl

### PR DESCRIPTION
**Pull request by mxmCherry**

### CODE REVIEW

File: ruby/bps-stan/lib/bps/stan.rb [hunk 1]
```diff
@@ -1,12 +1,55 @@
...
+  module Subscriber
+    register('stan') do |url, **opts|
+      cluster_id, client_id, url_opts = ::BPS::STAN::Utils.parse_url(url)
+      url_opts.update(opts)
+      STAN.new(cluster_id, client_id, **::BPS::STAN::Utils.coercer.coerce(url_opts))
+    end
+  end
+
+  module STAN
+    module Utils
```
Comment (mxmCherry) [inline ruby/bps-stan/lib/bps/stan.rb:23]: Not sure if this file is the best place to keep these common utils, but they're, well, really **common**

File: ruby/bps-stan/lib/bps/subscriber/stan.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        # handle TLS if CA file is provided:
+        if !nats[:tls] && nats[:tls_ca_file]
+          ctx = OpenSSL::SSL::SSLContext.new
+          ctx.set_params
+          ctx.ca_file = nats.delete(:tls_ca_file)
+          nats[:tls] = ctx
+        end
+
+        @client = ::STAN::Client.new
+        @client.connect(cluster_id, client_id, nats: nats, **opts.slice(*::BPS::STAN::Utils::CLIENT_OPTS.keys))
```
Comment (mxmCherry) [inline ruby/bps-stan/lib/bps/subscriber/stan.rb:19]: Maybe entire connection creating could be moved to smth like `@client = ...::Utils.connect(...)` or so? Right now it's just copy-paste.

Comment (dim) [reply ruby/bps-stan/lib/bps/subscriber/stan.rb:19]: yeah, would be nice, not sure if you need extra `Utils::` namespace either, just:

```
::BPS::STAN.connect # => ::STAN::Client
::BPS::STAN::CLIENT_OPTS # => { ... }
::BPS::STAN.coercer # => ...
```

**Review by dim — APPROVED** [review]

**Comment by dim** [comment]
Done!

**Pull request merged**